src/WebServer/Rest/UniqueMatchUriAndVerbComparer.cs

```diff
@@ -0,0 +1,44 @@
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Restup.Webserver.Rest
+{
+    internal class UniqueMatchUriAndVerbRestControllerMethodInfoComparer : IEqualityComparer<RestControllerMethodInfo>
+    {
+        private readonly PathPartsAndParametersParsedUriComparer _parsedUriComparer;
+
+        public UniqueMatchUriAndVerbRestControllerMethodInfoComparer()
+        {
+            _parsedUriComparer = new PathPartsAndParametersParsedUriComparer();
+        }
+
+        public bool Equals(RestControllerMethodInfo x, RestControllerMethodInfo y)
+        {
+            return _parsedUriComparer.Equals(x.MatchUri, y.MatchUri) && x.Verb == y.Verb;
+        }
+
+        public int GetHashCode(RestControllerMethodInfo obj)
+        {
+            unchecked
+            {
+                return ((obj.MatchUri?.GetHashCode() ?? 0) * 397) ^ (int)obj.Verb;
```

Comment:
Well this is a generally known standard way of creating an aggregated hash code in .net, you just need a prime to make it so that the distribution of hash codes is even enough. When I generated them with Resharper it used 397. See http://stackoverflow.com/questions/102742/why-is-397-used-for-resharper-gethashcode-override.

Unsure what to call the variable, would HashCodePrime suffice?